src/purifier.js

```diff
@@ -348,6 +349,19 @@ export function addPurifyHooks(purifier, diffing) {
       delete allowedTags[tag];
     });
     allowedTagsChanges.length = 0;
+
+    // Output user errors for each removed attribute or element.
+    this.removed.forEach(r => {
+      if (r.attribute) {
+        const {name, value} = r.attribute;
+        user().error(TAG, `Removed unsafe attribute: ${name}="${value}"`);
+      } else if (r.element) {
+        const {nodeName} = r.element;
+        if (nodeName !== 'REMOVE') { // <remove> is added by DOMPurify.
```

Comment:
why not just combine nodeName check in the initial element check.

    if (r.attribute) {
       const {name, value} = r.attribute;
       user().error(TAG, `Removed unsafe attribute: ${name}="${value}"`);
    } else if (r.element && r.element.nodeName !== 'REMOVE') {
    ...
    }